WordPress/Classes/Services/PostService.m

```diff
@@ -795,23 +794,11 @@ - (void)updatePost:(AbstractPost *)post withRemotePost:(RemotePost *)remotePost
             NSDictionary *publicizeMessageDictionary = [self dictionaryWithKey:@"_wpas_mess" inMetadata:remotePost.metadata];
             publicizeMessage = [publicizeMessageDictionary stringForKey:@"value"];
             publicizeMessageID = [publicizeMessageDictionary stringForKey:@"id"];
-
-            NSArray *disabledPublicizeConnectionsArray = [self entriesWithKeyLike:@"_wpas_skip_*" inMetadata:remotePost.metadata];
-            for (NSDictionary *disabledConnectionDictionary in disabledPublicizeConnectionsArray) {
-                NSString *dictKey = [disabledConnectionDictionary stringForKey:@"key"];
-                // We only want to keep the keyringID value from the key
-                NSNumber *keyringConnectionID = @([[dictKey stringByReplacingOccurrencesOfString:@"_wpas_skip_"
-                                                                                      withString:@""]integerValue]);
-                NSMutableDictionary *keyringConnectionData = [NSMutableDictionary dictionaryWithCapacity:2];
-                keyringConnectionData[@"id"] = [disabledConnectionDictionary stringForKey:@"id"];
-                keyringConnectionData[@"value"] = [disabledConnectionDictionary stringForKey:@"value"];
-                disabledPublicizeConnections[keyringConnectionID] = keyringConnectionData;
-            }
         }
         postPost.publicID = publicID;
         postPost.publicizeMessage = publicizeMessage;
         postPost.publicizeMessageID = publicizeMessageID;
-        postPost.disabledPublicizeConnections = disabledPublicizeConnections;
+        postPost.disabledPublicizeConnections = [self disabledPublicizeConnectionsForPost:post andMetadata:remotePost.metadata];
```

Comment:
I seem to be running into a crash with this change. I believe it's attempting to cast an array object in the `metadata` to a string.

<details><summary>Screenshot</summary>

![Crash](https://github.com/wordpress-mobile/WordPress-iOS/assets/2454408/57c585e8-7166-4d4f-b787-60d89f4faf75)

</details>

Log message:
```
Could not cast value of type '__NSSingleObjectArrayI' (0x136102b50) to 'NSString' (0x1ba228a38).
```

Metadata of the remote post:
```
[
{
    id = 16;
    key = "sharing_disabled";
    value =     (
        "<null>"
    );
},
{
    id = 17;
    key = "switch_like_status";
    value =     (
        "<null>"
    );
},
{
    id = 83;
    key = "_thumbnail_id";
    value = 11;
}
]
```